test/tools/cli/IonProcess.java

```diff
@@ -630,37 +593,33 @@ private static Event ionStreamToEvent(EventType eventType,
         SymbolToken fieldName = ionReader.getFieldNameSymbol();
         SymbolToken[] annotations = ionReader.getTypeAnnotationSymbols();
 
-        String valueText = null;
-        byte[] valueBinary = null;
+        IonValue value = null;
         if (eventType == EventType.SCALAR) {
             try (
                     ByteArrayOutputStream textOut = new ByteArrayOutputStream();
-                    IonWriter textWriter = IonTextWriterBuilder.standard().build(textOut);
-                    ByteArrayOutputStream binaryOut = new ByteArrayOutputStream();
-                    IonWriter binaryWriter = IonBinaryWriterBuilder.standard().build(binaryOut);
-                    ) {
+                    IonWriter tempWriter = IonTextWriterBuilder.standard().build(textOut);
+            ) {
                 //write Text
-                textWriter.writeValue(ionReader);
-                textWriter.finish();
-                valueText = textOut.toString("utf-8");
-                //write binary
-                binaryWriter.writeValue(ionReader);
-                binaryWriter.finish();
-                valueBinary = binaryOut.toByteArray();
+                tempWriter.writeValue(ionReader);
+                tempWriter.finish();
+                String valueText = textOut.toString("utf-8");
+                String[] s = valueText.split("::");
+                value = IonSystemBuilder.standard().build().singleValue(s[s.length -1]);
             }
         }
 
         ImportDescriptor[] imports = null;
         int depth = ionReader.getDepth();
-        return new Event(eventType, ionType, fieldName, annotations, valueText, valueBinary, imports, depth);
+        return new Event(eventType, ionType, fieldName, annotations, value, imports, depth);
     }
 
     private static boolean isSameSymbolTable(SymbolTable newTable, SymbolTable curTable) {
         if (newTable == null && curTable == null) return true;
         else if (newTable != null && curTable == null) return false;
         else if (newTable == null) return false;
-
-        if (newTable.isLocalTable() && newTable.getImportedTables().length == 0) {
+        if (newTable.isSystemTable() && newTable.getName() == "$ion") {
```

Comment:
Why isn't this `if (newTable.isSystemTable() && newTable.isSystemTable())`?

Reply:
Changed.

question:
what's system table's version. I think for $ion we use 0 right?